skbio/metadata/_mixin.py

```diff
@@ -406,3 +407,157 @@ def has_positional_metadata(self):
         """
         return (self._positional_metadata is not None and
                 len(self.positional_metadata.columns) > 0)
+
+
+class IntervalMetadataMixin(metaclass=abc.ABCMeta):
+    @abc.abstractmethod
+    def _interval_metadata_axis_len_(self):
+        ''''''
+        raise NotImplementedError
+
```

Comment:
pls c.f. `PositionalMetadataMixin`. basically the child class inheriting the mixins handle that in its own polymorphismi way